Parser.pm6

```diff
@@ -65,23 +65,23 @@ class pCNI is Hash {
 
 	
 	# returns kv pairs of all keys in a section, with the section stripped
-	method SubRec($prefix) {
+	method SubTree($prefix) {
 		return self.sub: $prefix, /./;
 	}
 
 	# returns kv pairs of all keys directly in a section, with the section stripped
-	method SubFlat($prefix) {
+	method SubLeaves($prefix) {
 		return self.sub: $prefix, /<-[.]>/;
 	}
 
 	# returns a list of all values in a section
-	method ListRec($prefix) {
-		return self.SubRec($prefix).values.list;
+	method ListTree($prefix) {
+		return self.SubTree($prefix).values.list;
 	}
 
 	# returns a list of all values directly in the section
-	method ListFlat($prefix) {
-		return self.SubFlat($prefix).values.list;
+	method ListLeaves($prefix) {
+		return self.SubLeaves($prefix).values.list;
 	}
 
 	# you are also recommended to implement WalkFlat and WalkRec alongside the above
```

Comment:
```suggestion
	# you are also recommended to implement WalkLeaves and WalkTree alongside the above
```
I don't know if `KeyLeaves` and `KeyTree` might also be interesting.

Reply:
Walker pattern gets called walk, surely.
It's not like it gives you a set of keys, eh?

Reply:
Yes, the walk functions stay as they are (apart from updating their names to be consistent as in my code suggestion). I meant KeyLeaves/KeyTree as *additional* API functions might be interesting, since we also have the List... functions.

Reply:
Oh, now I see.
I think that's interesting, as well as SectionTree and SectionLeaves.
I don't want to require too much from library writers though.
Then again, the API is purely a suggestion / informative...